### ISSUE TITLE
Allow selectively skipping elements so that the "driver-fix-stacking" class won't get applied

### ISSUE THREAD

**Issue by josias-r** [kamranahmedse/driver.js#346]
There are just some cases where you want to keep zindex on a parentNode or other styles. As a dev I want to be able therefore mark an element as `driver-dont-fix-stacking` or something like that.

This way a dev can choose to opt-out for specific elements that their element will be ignored of being reset by the `driver-fix-stacking`.

It would be very simple to implement this option. Just add another check that wraps the current checks in [this file](https://github.com/kamranahmedse/driver.js/blob/84b6ab2a586f572062fbe1f350c4034b8a17b605/src/core/element.js#L265).

Wrapping the if in something like this:
```js
if (!parentNode.classList.contains(CLASS_IGNORE_STACKING_CONTEXT_FIX)) {
   // ... if should add fix class
}
```

This way any element with that `CLASS_IGNORE_STACKING_CONTEXT_FIX` will keep all it's original styles.

At the moment a similar result can be achieved by overwriting the `driver-dont-fix-stacking` class styles for specific elements which is very cumbersome to work with. So being able to prevent the class being added in the first place would be amazing :)

**Edit:**
It turns out to be a fundamental flaw on how driver.js is designed. So [I ported the library and refactored it](https://josias-r.github.io/boarding.js/#comparison-table) + added more functionally which I needed.

**Comment by josias-r** [comment]
For the moment, I wrote a react-hook for our codebase to do exactly that. It simply watches an element for class changes and if it is the `driver-dont-fix-stacking` class, it will immediately remove it again:

```ts
import { useEffect, useRef } from "react";
// import { CLASS_FIX_STACKING_CONTEXT } from "driver.js/src/common/constants"; // not property exported, but this is where it comes from
const CLASS_FIX_STACKING_CONTEXT = "driver-fix-stacking";

function useDriverJSIgnoreStackFix<TElement extends HTMLElement>() {
  const elementRef = useRef<TElement>(null);
  useEffect(() => {
    const observer = new MutationObserver((mutations) => {
      mutations.forEach((mutation) => {
        if (
          mutation.type !== "attributes" &&
          mutation.attributeName !== "class"
        )
          return;
      });

      if (elementRef.current?.classList.contains(CLASS_FIX_STACKING_CONTEXT)) {
        console.log("REMOVING!!");

        elementRef.current.classList.remove(CLASS_FIX_STACKING_CONTEXT);
      }
    });

    if (elementRef.current) {
      observer.observe(elementRef.current, {
        attributes: true,
        attributeFilter: ["class"],
      });
    } else {
      console.warn(
        "Element that was supposed to be watched has not been assigend yet"
      );
    }

    return () => {
      observer.disconnect();
    };
  }, []);
  return elementRef;
}

export default useDriverJSIgnoreStackFix;
```

You can use it like this:
```tsx
function MyComponent() {
   const ignoreDriverStackFixRef = useDriverJSIgnoreStackFix<HTMLDivElement>();

   return <div className="myClassesThatWouldTriggerStackFixClass" ref={ignoreDriverStackFixRef}>children</div>
}
```

**Comment by josias-r** [comment]
After more extensive usage testing (we are just now working on a prototype in our application that makes use of this library), I realised that there is a more fundamental flaw of how this library handles highlighting of elements.

The reason `driver-dont-fix-stacking` is necessary in the first place, is because the library wants to force a zIndex on the element that should be highlighted. It needs to do that, so it can put that element-to-highlight on a higher index then the overlay that appears.

Now the issue is, that this might work fine for many simple websites. But for more complex websites or web-apps. There's a high chance the complex layout just relies on zIndex (for example modals/dialogs that open, etc.).

So what would be the solution? Actually the solution would be to work without touch the stack context at all and leaving everything as it is.
This would mean though, that you cannot make use of the "overlay" anymore, as it would block clicking the element.

Therefore we are now experimenting in a custom solution where the highlight works using an SVG where there is an actual rectangle cropped out. So far the results seem to be very pleasing.

This means we can patch out all of the stack-context classes that driver.js applies to the elements like so:
```js
document
  .querySelectorAll(
    ".driver-fix-stacking,.driver-highlighted-element,.driver-position-relative"
  )
  .forEach((node) => {
    node.classList.remove("driver-fix-stacking");
    node.classList.remove("driver-highlighted-element");
    node.classList.remove("driver-position-relative");
  });
```

@kamranahmedse Did you every think of going into that direction? Our results so far seem to be promising and we don't see a big issue with this. If you like the idea I can also work on a PR to get this into `driver.js`.
If no other issues start appearing, this approach seems much more stable, as driver.js can really "highlight" an element without having to hack the elements zIndex to the top and potentially breaking layouts.

**Comment by kamranahmedse** [comment]
Thank you for looking into this @josias-r. The SVG approach does seem interesting. The initial implementation of driver.js was similar to what you proposed -- we used to draw a canvas covering the full page and were cutting the portion above the highlighted element but the [approach was discarded](https://github.com/kamranahmedse/driver.js/pull/10) due to browser compatibility and performance issues.

I just got some time and have started working on the rewrite. I will look into what you have done with the fork :)

**Comment by kamranahmedse** [comment]
Thanks @josias-r — I just released [v1.0](https://github.com/kamranahmedse/driver.js/releases/tag/1.0.3) of driver.js which has been rewritten from the ground up using the SVG mechanism as your boarding.js. Please have a [look at the docs](https://driverjs.com/) and do let me know if you face any issues. 

Once again, thank you for your contribution 🙏

**Comment by josias-r** [comment]
@kamranahmedse Interesting! I'm currently off from my computer for a few months, but glad my rewrite was an inspiration 😀 

I will check it out again at some point
